Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -864,67 +864,70 @@ public int damage(int damage, UUID attackerId, Ability source, Game game, boolea
      */
     private int doDamage(int damageAmount, UUID attackerId, Ability source, Game game, boolean preventable, boolean combat, boolean markDamage, List<UUID> appliedEffects) {
         int damageDone = 0;
-        if (damageAmount > 0 && canDamage(game.getObject(attackerId), game)) {
-            if (this.isPlaneswalker()) {
-                damageDone = damagePlaneswalker(damageAmount, attackerId, source, game, preventable, combat, markDamage, appliedEffects);
+        if (damageAmount <1 || !canDamage(game.getObject(attackerId), game)) {
+            return damageDone;
+        }
+        if (this.isPlaneswalker()) {
+            damageDone = damagePlaneswalker(damageAmount, attackerId, source, game, preventable, combat, markDamage, appliedEffects);
+        }
+        else {
```

Comment:
I saw bug report from #7591 about creature + planeswalker damage problem in [[Gideon, Battle-Forged]]. So there is must be both damage calls, not `else`?

Reply:
> change how permanents handle being damaged

Never mind, it's not yet done.